Sources/Basics/Observability.swift

```diff
@@ -307,6 +326,40 @@ public struct Diagnostic: CustomStringConvertible, Equatable {
     }
 }
 
+// MARK: - Metrics
+
+public protocol MetricsHandler {
+    func handleTimer(scope: ObservabilityScope, label: String, duration: DispatchTimeInterval)
+}
+
+public struct Timer: TimerProtocol {
```

Comment:
I'm not sure this is a good name since `Timer` in the everyday world is usually more like an alarm clock or some kind of callback.  I don't have a good candidate name in mind though.